perception/sandbox/polygrasp/src/polygrasp/cam_pub_sub.py

```diff
@@ -0,0 +1,126 @@
+import logging
+import json
+from types import SimpleNamespace
+
+import numpy as np
+import open3d as o3d
+
+import a0
+import realsense_wrapper
+from polygrasp import serdes
+
+log = logging.getLogger(__name__)
+topic = "cams/rgbd"
+
+
+class CameraSubscriber:
+    def __init__(self, intrinsics_file, extrinsics_file):
+        with open(intrinsics_file, "r") as f:
+            intrinsics_json = json.load(f)
+            self.intrinsics = [SimpleNamespace(**d) for d in intrinsics_json]
+
+        with open(extrinsics_file, "r") as f:
+            self.extrinsics = json.load(f)
+
+        self.sub = a0.SubscriberSync(a0.PubSubTopic(topic), a0.INIT_MOST_RECENT)
+
+    def get_rgbd(self):
+        return serdes.bytes_to_np(self.sub.read().payload)
```

Comment:
With the `SubscriberSync` set up with `a0.INIT_MOST_RECENT`, the first read will be "latched", whatever is the most-recent element in the topic. After that, it iterates sequentially.

If you always want the most recent, you'll want to add `a0.ITER_NEWEST`.

Also, be careful. `read` will never return the same packet twice. You should check if a newer packet is available with `can_read` and maybe cache the most recent yourself.

```py
self.sub = a0.SubscriberSync(topic, a0.INIT_MOST_RECENT, a0.ITER_NEWEST)
self.recent_rgbd = None

def get_rgbd(self):
    if self.sub.can_read():
        self.recent_rgbd = serdes.bytes_to_np(self.sub.read().payload)
    return self.recent_rgbd
```